### DISCUSSION TITLE
Strange regex for Internal link alias transformation

### DISCUSSION THREAD

**Discussion by UserBoi1** [tadashi-aikawa/obsidian-various-complements-plugin#218] [Q&A]
Hi there,

I am trying to understand the alias transformation feature. I want to use it to give me uncapitalised aliases. All my note names start with capitals (like Wikipedia pages), but this means whenever I use autocomplete, it gives me the capitalised name, when I want it in plain lowercase in the text.

I have read a fair amount online and I have found lots of resources. But none of them look at all like this regex. There's named variables with angle brackets, which I can't find anything about online, some kind of `(?regex)` grouping potentially (all I know is things don't seem to work without it), and it doesn't recognise `\l` as a reserved regex thing, despite the fact everyone online suggests using it for lowercase.

Does anyone know what kind of regex this is and how I might achieve my goal of lowercase aliases?

**Comment by tadashi-aikawa** [comment]
Hi, @UserBoi1 :)

I have no idea as far, but I'll survey the solution to solve your problems 👍

**Reply by UserBoi1** [reply]
Thank you!

**Reply by tadashi-aikawa** [reply]
@UserBoi1 
I have investigated it, but I found that JavaScript/TypeScript doesn't support `\l` or `\L`.

> All my note names start with capitals (like Wikipedia pages), but this means whenever I use autocomplete, it gives me the capitalised name, when I want it in plain lowercase in the text.

I think your suggestion is used for general purposes. So if you want the feature, please create a new issue (FR) with a few examples.

Kind regards.

**Reply by UserBoi1** [reply]
Thanks, I will add one
